### DISCUSSION TITLE
Problem with authentication (Oauth) redirect

### DISCUSSION THREAD

**Discussion by RobertRacklBMW** [cypress-io/cypress#29469] [Questions and Help]
We have a web application that uses Oauth 2.0. Authentication happens via Oauth Implicit Flow. That means. The user is redirected to a login page. There he enteres username and password. And then he is redirected back to our application.

I am trying to write a Cypress test case for this. I did already read a lot about cypress.origin() command and more. But I still cannot get it to work.

````
  cy.visit("/frontend-web-app")  
  // browser is now automatically forwarded to our-company.com/auth  (HTTP 301)
  // in the cypress test the URL at the very top (not in my app, but the cypress browser window changes!!!)

  cy.get('#username').type(username)
  cy.get('#password').type(pass)
  cy.get('#loginButton').click()
	
  // now the browser is redirected back to to our app. Cypress also shows our app. Successfully logged in.
  // BUT THIS FAILS ????
  cy.url().should('equal', "/frontend-web-app")

  // the URL at the very top of the cypress browser still shows  our-company.com/auth   and not  /frontend-web-app   ????
````

The version at the top does not use cy.origin() at all. Login does work. But somehow the redirect back does not fully work. Any ideas?

**Comment by portyu9** [comment]
Hi @RobertRacklBMW 

One issue in the example is this assertion:

```js
cy.url().should('equal', '/frontend-web-app')
```

`cy.url()` returns the **full URL**, not just the path. If you only want to verify the redirect path, use:

```js
cy.location('pathname').should('eq', '/frontend-web-app')
```

Regarding `cy.origin()`: `/auth` and `/frontend-web-app` are not different origins if the scheme, hostname, and port are unchanged. They are only different paths.

If authentication actually navigates to a different hostname, scheme, or port, then `cy.origin()` is required with current Cypress.

If `cy.location('pathname')` still reports `/auth` after login, then the browser has not actually navigated back to `/frontend-web-app`, so I would inspect the OAuth callback/redirect rather than treating it as a Cypress URL-display issue.

Please mark this answer as accepted if it helped, thank you.
